app/models/schedule.rb

```diff
@@ -17,9 +17,17 @@ class Schedule < ApplicationRecord
   belongs_to :cohort
   has_many :courses, through: :course_group
 
+  delegate :editable?, to: :cohort
+
+  normalizes :allowed_declaration_types, with: ->(value) { value.reject(&:blank?) }
+
   validates :name, presence: true
   validates :identifier, presence: true, uniqueness: { scope: :cohort_id }
   validates :ecf_id, uniqueness: { case_sensitive: false }, allow_nil: true
   validates :applies_from, presence: true
   validates :applies_to, presence: true
+
+  def self.allowed_declaration_types
+    connection.execute("SELECT unnest(enum_range(NULL::declaration_types))").values.flatten
+  end
```

Comment:
This seems a bit unusual or I'm not following? Whats the advantage of normalizing the list in a separate method, why not normalize it here? Though I'm also not understanding why postgres would return a blank enum type?